src/features/locators/components/LocatorMenu.tsx

```diff
@@ -15,132 +16,193 @@ import {
   upPriority,
   copyLocatorOption,
   dividerItem,
+  addToPO,
+  removeFromPO,
 } from "../../../common/components/menu/menuOptions";
-import { isProgressStatus, locatorGenerationController } from "../utils/locatorGenerationController";
+import { locatorGenerationController } from "../utils/locatorGenerationController";
 import {
   Locator,
   LocatorCalculationPriority,
   LocatorTaskStatus,
   ValidationStatus,
   LocatorValidationWarnings,
 } from "../types/locator.types";
-import { setCalculationPriority, toggleDeleted } from "../locators.slice";
+import {
+  setCalculationPriority,
+  toggleDeleted,
+  toggleDeletedGroup,
+  toggleElementGroupGeneration,
+} from "../locators.slice";
 import { getCopyOptions, getLocatorValidationStatus } from "../utils/utils";
 import { rerunGeneration } from "../reducers/rerunGeneration.thunk";
+import { stopGenerationGroup } from "../reducers/stopGenerationGroup.thunk";
 import { stopGeneration } from "../reducers/stopGeneration.thunk";
 import { LocatorType } from "../../../common/types/common";
 import { useSelector } from "react-redux";
 import {
   selectCalculatedActiveByPageObj,
-  selectCurrentPageObject,
   selectWaitingActiveByPageObj,
+  selectActiveGenerateByPO,
+  selectActiveNonGenerateByPO,
+  selectDeletedActiveByPageObj,
+  selectFailedSelectedByPageObject,
+  selectInProgressActiveByPageObject,
 } from "../../../features/pageObjects/pageObject.selectors";
 import { OnboardingContext } from "../../onboarding/OnboardingProvider";
 
 interface Props {
-  element: Locator;
+  element?: Locator;
```

Comment:
it can be removed from Props if you don't use it inside component

Reply:
done